kosmorrolib/events.py

```diff
@@ -528,3 +528,105 @@ def get_events(for_date: date = date.today(), timezone: int = 0) -> [Event]:
         end_date = date(end_date.year, end_date.month, end_date.day)
 
         raise OutOfRangeDateError(start_date, end_date) from error
+
+
+def search_events(
+    event_types: [EventType], end: date, start: date = date.today(), timezone: int = 0
+) -> [Event]:
+    """Search between `start` and `end` dates, and return a list of matching events for the given time range, adjusted to a given timezone.
+
+    Find all events between January 27th, 2020 and January 29th, 2020:
+
+    >>> event_types = [EventType.OPPOSITION, EventType.CONJUNCTION, EventType.OCCULTATION, EventType.MAXIMAL_ELONGATION, EventType.APOGEE, EventType.PERIGEE, EventType.SEASON_CHANGE, EventType.LUNAR_ECLIPSE]
+    >>> search_events(event_types, date(2020, 1, 29), date(2020, 1, 27)) # doctest: +NORMALIZE_WHITESPACE
```

Comment:
I think it might be more readable to use named parameters for `start` and `end` for documentation purpose, since they are inverted in the signature due to the default `start` value.
WDYT?

Reply:
I agree, that's a good idea. It event took me a minute to get this test to pass because I had them backwards at first :laughing: